cmd/influxd/restore/restore.go

```diff
@@ -309,7 +309,8 @@ func (cmd *Command) unpackTar(tarFile string) error {
 
 // unpackFile will copy the current file from the tar archive to the data dir
 func (cmd *Command) unpackFile(tr *tar.Reader, fileName string) error {
-	fn := filepath.Join(cmd.datadir, fileName)
+	nativeFileName := filepath.FromSlash(fileName)
```

Comment:
Yes, this is backwards compatible since FromSlash called with a filename with backslashes will not find any slashes to convert. To be sure I wrote a little test program to make sure I understood the docs correctly.

On Windows
```
Input:     c:\foo\bar
ToSlash:   c:/foo/bar
FromSlash: c:\foo\bar

Input:     /foo/bar
ToSlash:   /foo/bar
FromSlash: \foo\bar
```

On Linux
```
Input:     c:\foo\bar
ToSlash:   c:\foo\bar
FromSlash: c:\foo\bar

Input:     /foo/bar
ToSlash:   /foo/bar
FromSlash: /foo/bar
```